src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const grouped: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const groupKey = item[key] as string;
```

Comment:
Casting the key to a string assumes that the key is always a string. However, the key could be of any type that is a property of T. Consider using a more robust method to handle keys of different types, such as converting the key to a string using `String(item[key])` or ensuring that the key is indeed a string type.